BCStudentSoftwareDevTeam/celts
# PR for Issue #30

This PR fixes Issue #30 

Summary:
 Created a section for a larger page that allows admins to add Volunteers and Outside Participants to an Event they're creating. We created the UI based on the Balsamic with minor changes along the way. We created a modal that allows the admin to create a participant and submit their information and append it to a table that will show all the Participants they have created for this event. They will also have the option to remove the Participant which will remove their information from the table and stop them from being added to the database. In the other section, we created a table that will hold all the volunteers that will be added to the table, we have provided a search bar that will allow the Admin to type and will send their input to the database and display the results that meet the criteria when the Name they are searching for is clicked on the Volunteer is then added to the table, again we have provided a remove button that will remove the Volunteer form the table. We also provided tests in a new folder found at `/tests/code/test_participants.py` these tests will check for some possible cases the Admin may present.

To Test: 
  To test you can run the test suites and see they all pass. You can also visit `http://<YOURIP>:8080/addParticipants` and view our UI. You can then click and search "Sc" in the search bar and see the results appear. You can also search "Br" and see no admins will appear. You can then click the "+" button and create a participant and see that the participant is then added to the table upon completion and submission of the form. You can then freely remove participants you have created and Volunteers you have added from their respective tables.

Files Created: 
`/app/templates/addParticipants.html`
`/app/static/js/addParticipants.js`
`/app/controllers/admin/addParticipants.py`
`/tests/code/test_participants.py` 

Notes: 
This PR upon acceptance will need to be merged with `create-event-issue28` and altered to allow submission of our section along with the other data on the page. We will also need to implement the required fields.
